File: src/test/java/io/plagov/advent/aoc2016/Day03Test.java
package io.plagov.advent.aoc2016;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import util.FileUtil;

import java.util.List;

class Day03Test {

    private final List<String> input = FileUtil.readInputFile("2016/day03.txt");

    @Test
    void solveFirst() {
        var day03 = new Day03();
        Assertions.assertEquals(993, day03.partOne(input));
    }

    @Test
    void solveSecond() {
        var day03 = new Day03();
        Assertions.assertEquals(1849, day03.partTwo(input));
    }
}

File: src/test/kotlin/io/plagov/advent/aoc2024/Day03Test.kt
package io.plagov.advent.aoc2024

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import util.readInputFileAsString

class Day03Test {

  private val input = readInputFileAsString("2024/day03.txt")

  private val day3 = Day03()

  @Test
  fun solveFirst() {
    Assertions.assertEquals(191183308, day3.partOne(input))
  }

  @Test
  fun solveSecond() {
    Assertions.assertEquals(92082041, day3.partTwo(input))
  }
}
